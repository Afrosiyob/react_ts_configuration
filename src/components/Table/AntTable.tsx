import React from "react";
import { Table } from "antd";

import { useSetKey } from "./hooks";
import { Types } from ".";

const AntTable: React.FC<Types.IProps> = ({
  columns,
  data,
  loading,
  handleTable,
  pagination,
  className,
}) => {
  const list: Types.ITableData[] = useSetKey(data);

  return (
    <Table
      data-testid="ANT_TABLE_ID"
      className={className}
      columns={columns}
      dataSource={list}
      pagination={pagination != null ? pagination : false}
      loading={loading}
      onChange={handleTable}
      scroll={{ x: "auto" }}
    />
  );
};

export default AntTable;
