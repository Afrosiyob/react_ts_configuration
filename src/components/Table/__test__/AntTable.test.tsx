/* eslint-disable @typescript-eslint/strict-boolean-expressions */
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import AntTable from "../AntTable";

describe("Test for component 'AntTable'", () => {
  beforeAll(() => {
    global.matchMedia =
      global.matchMedia ||
      function () {
        return {
          addListener: jest.fn(),
          removeListener: jest.fn(),
        };
      };
  });

  it("should render", () => {
    const columns = [
      {
        key: 0,
        dataIndex: "name",
        title: "Name",
      },
      {
        key: 1,
        dataIndex: "age",
        title: "Age",
      },
      {
        key: 2,
        dataIndex: "address",
        title: "Address",
      },
    ];

    const data = [
      {
        name: "Leo",
        age: 19,
        address: "London",
      },
      {
        name: "Doe",
        age: 25,
        address: "America",
      },
    ];

    render(<AntTable columns={columns} data={data} loading={false} />);

    const component = screen.getByTestId("ANT_TABLE_ID");

    expect(component).toBeInTheDocument();
  });

  it("should empty", () => {
    const columns = [
      {
        key: 0,
        dataIndex: "name",
        title: "Name",
      },
      {
        key: 1,
        dataIndex: "age",
        title: "Age",
      },
      {
        key: 2,
        dataIndex: "address",
        title: "Address",
      },
    ];

    render(<AntTable columns={columns} data={[]} loading={false} />);

    const component = screen.getByTestId("ANT_TABLE_ID");

    expect(component).toHaveTextContent("NameAgeAddress");
  });
});

describe("Snapshot for component 'AntTable'", () => {
  it("should match DOM", () => {
    const columns = [
      {
        key: 0,
        dataIndex: "name",
        title: "Name",
      },
      {
        key: 1,
        dataIndex: "age",
        title: "Age",
      },
      {
        key: 2,
        dataIndex: "address",
        title: "Address",
      },
    ];

    const data = [
      {
        name: "Leo",
        age: 19,
        address: "London",
      },
      {
        name: "Doe",
        age: 25,
        address: "America",
      },
    ];

    const tree = renderer.create(
      <AntTable columns={columns} data={[]} loading={false} />
    );

    expect(tree).toMatchSnapshot();
  });
});
