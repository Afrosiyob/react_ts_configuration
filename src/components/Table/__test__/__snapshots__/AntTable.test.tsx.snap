// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for component 'AntTable' should match DOM 1`] = `
<div
  className="ant-table-wrapper css-dev-only-do-not-override-1n7nwfa"
>
  <div
    className="ant-spin-nested-loading css-dev-only-do-not-override-1n7nwfa"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-empty ant-table-scroll-horizontal"
        data-testid="ANT_TABLE_ID"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={
              Object {
                "overflowX": "auto",
                "overflowY": "hidden",
              }
            }
          >
            <table
              style={
                Object {
                  "minWidth": "100%",
                  "tableLayout": "auto",
                  "width": "auto",
                }
              }
            >
              <colgroup />
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={Object {}}
                  >
                    Age
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    scope="col"
                    style={Object {}}
                  >
                    Address
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  aria-hidden="true"
                  className="ant-table-measure-row"
                  style={
                    Object {
                      "fontSize": 0,
                      "height": 0,
                    }
                  }
                >
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                  <td
                    style={
                      Object {
                        "border": 0,
                        "height": 0,
                        "padding": 0,
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "hidden",
                        }
                      }
                    >
                       
                    </div>
                  </td>
                </tr>
                <tr
                  className="ant-table-placeholder"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                >
                  <td
                    className="ant-table-cell"
                    colSpan={3}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <div
                      className="ant-table-expanded-row-fixed"
                      style={
                        Object {
                          "left": 0,
                          "overflow": "hidden",
                          "position": "sticky",
                          "width": 0,
                        }
                      }
                    />
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
